users/user-guide-query/multi-stage-query/operator-types/hash_join.md

```diff
@@ -0,0 +1,210 @@
+  values for the columns from the right relation.
+- Right join, like the left join but returning all the rows from the right relation, with the columns from the left 
+  relation filled with null values for the rows that do not have a match.
+- Full outer join, where all the rows from both relations are returned. If a row from any relation does not have a match 
+  in the other relation, the columns from the other relation are filled with null values.
+- Semi-join, where only the rows from the left relation that have a match in the right relation are returned. This is 
+  useful to filter the rows from the left relation based on the existence of a match in the right relation.
+- Anti-join, where only the rows from the left relation that do not have a match in the right relation are returned.
+
+## Implementation details
+
+The hash join operator is one of the new operators introduced in the multi-stage query engine.
+The current implementation assumes that the right input relation is the smaller one, so it consumes this
+input first building a hash table that is then probed with the left input relation.
+
+{% hint style="info" %}
+Future optimizations may include advanced heuristics to decide which input relation to consume first, but in the current
+implementation, it is important to specify the smaller relation as the right input.
+{% endhint %}
+
+Although the whole multi-stage query engine is designed to be able to process the data in memory, the multi-stage query
+engine uses the ability to execute each stage in different workers (explained in 
+[understanding stages](../understanding-stages.md)) to be able to process the data that may not fit in the memory of a 
+single node. Specifically, each worker processes a subset of the data. Inputs are by default partitioned by the join 
+keys and each worker process one partition of the data.
+
+This means that data usually needs to be shuffled between workers, which is done by the engine using a mailbox system.
+The engine tries to minimize the amount of data that needs to be shuffled by partitioning the data, but some techniques
+can be used to reduce the amount of data that needs to be shuffled, like using co-located joins.
+
+### Blocking nature
+The hash join operator is a blocking operator. It needs to consume all the input data (from both inputs) before emitting
+the result.
+
+### Maximum number of groups
+Even using partitioning, the amount of data that needs to be stored in memory can be high, so the engine tries to
+protect itself from running out of memory by limiting the number of groups that can be created by the join keys.
+
+The [join_overflow_mode](../hints/join_overflow_mode.md) hint can be used to control the behavior of the engine when the
+number of groups exceeds the limit.
+This limit can be defined using the [max_rows_in_join](../hints/max_rows_in_join.md) hint.
+By default, this limit is slightly above 1 million groups and the default join overflow mode is `THROW`,
```

Comment:
why default behavior to throw in join, but in aggregate to just skip with partial?